jcl/src/java.base/share/classes/openj9/internal/foreign/abi/InternalDowncallHandler.java

```diff
@@ -358,9 +378,20 @@ private final long memSegmtOfPtrToLongArg(MemorySegment argValue, LinkerOptions
 			/* Store the heap argument's base object and offset. */
 			AbstractMemorySegmentImpl segment = (AbstractMemorySegmentImpl)argValue;
 			info.append(segment.unsafeGetBase(), segment.unsafeGetOffset());
+			MemorySessionImpl session = segment.sessionImpl();
+			if (session != null) {
+				session.acquire0();
+				info.addSession(session);
+			}
 		} else {
 			/* Set null to the heap base object to denote a native array. */
 			info.append(null, 0L);
+			AbstractMemorySegmentImpl segment = (AbstractMemorySegmentImpl)argValue;
+			MemorySessionImpl session = segment.sessionImpl();
+			if (session != null) {
+				session.acquire0();
+				info.addSession(session);
+			}
 		}
```

Comment:
With this change, `segment` is used in both branches of the `if` statement and the code to handle sessions is identical; suggesting moving `segment` before the `if` statement, the session handling after it:
```java
		/*[IF JAVA_SPEC_VERSION >= 22]*/
		AbstractMemorySegmentImpl segment = (AbstractMemorySegmentImpl)argValue;
		if (!argValue.isNative() && options.allowsHeapAccess()) {
			/* Store the heap argument's base object and offset. */
			info.append(segment.unsafeGetBase(), segment.unsafeGetOffset());
		} else {
			/* Set null to the heap base object to denote a native array. */
			info.append(null, 0L);
		}
		MemorySessionImpl session = segment.sessionImpl();
		if (session != null) {
			session.acquire0();
			info.addSession(session);
		}
		/*[ENDIF] JAVA_SPEC_VERSION >= 22 */
```